src/main/java/com/percero/agents/sync/services/DAODataProvider.java

```diff
@@ -0,0 +1,1279 @@
+							String nextKey = RedisKeyUtils.classIdPair(fieldObject.getClass().getCanonicalName(), ((IPerceroObject)fieldObject).getID());
+							keysToDelete.add(nextKey);
+						}
+						else if (fieldObject instanceof Collection) {
+							Iterator<Object> itrFieldObject = ((Collection) fieldObject).iterator();
+							while(itrFieldObject.hasNext()) {
+								Object nextListObject = itrFieldObject.next();
+								if (nextListObject instanceof IPerceroObject) {
+									String nextKey = RedisKeyUtils.classIdPair(nextListObject.getClass().getCanonicalName(), ((IPerceroObject)nextListObject).getID());
+									keysToDelete.add(nextKey);
+								}
+							}
+						}
+					}
+				}
+				Iterator<MappedFieldPerceroObject> itrToOneFields = mappedClass.toOneFields.iterator();
+				while(itrToOneFields.hasNext()) {
+					MappedFieldPerceroObject nextMappedField = itrToOneFields.next();
+					Object fieldObject = nextMappedField.getGetter().invoke(perceroObject);
+					if (fieldObject != null) {
+						if (fieldObject instanceof IPerceroObject) {
+							String nextKey = RedisKeyUtils.classIdPair(fieldObject.getClass().getCanonicalName(), ((IPerceroObject)fieldObject).getID());
+							keysToDelete.add(nextKey);
+						}
+						else if (fieldObject instanceof Collection) {
+							Iterator<Object> itrFieldObject = ((Collection) fieldObject).iterator();
+							while(itrFieldObject.hasNext()) {
+								Object nextListObject = itrFieldObject.next();
+								if (nextListObject instanceof IPerceroObject) {
+									String nextKey = RedisKeyUtils.classIdPair(nextListObject.getClass().getCanonicalName(), ((IPerceroObject)nextListObject).getID());
+									keysToDelete.add(nextKey);
+								}
+							}
+						}
+					}
+				}
+				
+				if (!keysToDelete.isEmpty()) {
+					cacheDataStore.deleteKeys(keysToDelete);
+					// TODO: Do we simply delete the key?  Or do we refetch the object here and update the key?
+					//redisDataStore.setValue(nextKey, ((BaseDataObject)perceroObject).toJson());
+				}
+			}
+			
+			return (T) cleanObject(perceroObject, userId);
+		}
+		catch(PropertyValueException pve) {
+			log.error("Error creating object", pve);
+			
+			SyncDataException sde = new SyncDataException(SyncDataException.MISSING_REQUIRED_FIELD, SyncDataException.MISSING_REQUIRED_FIELD_CODE, "Missing required field " + pve.getPropertyName());
+			sde.fieldName = pve.getPropertyName();
+			throw sde;
+		}
+		catch(Exception e) {
+			log.error("Error creating object", e);
+			
+			SyncDataException sde = new SyncDataException(SyncDataException.CREATE_OBJECT_ERROR, SyncDataException.CREATE_OBJECT_ERROR_CODE);
+			throw sde;
+		}
+	}
+	
+
+	////////////////////////////////////////////////////
+	//	PUT
+	////////////////////////////////////////////////////
+	@SuppressWarnings("unchecked")
+	public <T extends IPerceroObject> T putObject(T perceroObject, Map<ClassIDPair, Collection<MappedField>> changedFields, String userId) throws SyncException {
```

Comment:
Another method with lots of nesting. Might wanna break it out into helper methods for the branches.